### ISSUE TITLE
How to identify query params with keys only and no value

### ISSUE THREAD

**Issue by jlmllorente** [tiangolo/fastapi#5424]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/events")
def get_alerts(request: Request)
    params = request.query_params
```

### Description

I want to handle a use case where I want to handle a use case where if a query param is passed but no value is set, I would return a specific message. I want a different behavior when to when it is not passed at all.

I tried using request.query_params but it doesn't get the Key in the request as well.

Postman request looks like this:
<img width="805" alt="image" src="https://user-images.githubusercontent.com/104721284/192010955-160c2418-63f3-46ac-9f64-a416b92c03ae.png">

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

3.9

**Comment by JarroVGIT** [comment]
If the query parameter has no value, it can be cast as an empty string `""`. See the following example:

```python
from fastapi import FastAPI, Query, Request

app = FastAPI()


@app.get("/")
async def root(r: Request, event: int | str | None = None, item: int | None = None):
    print(f"Event:{event}.")
    print(f"Item:{item}.")
    return {"event": event, "item": item}


if __name__ == "__main__":
    import uvicorn

    uvicorn.run(app, host="0.0.0.0", port=8000)
```

Following requests with responses (note the null value!):
```bash
 % curl 'localhost:8000/?item=1'
{"event":null,"item":1}%                          
                    
 % curl 'localhost:8000/?item=1&event'
{"event":"","item":1}%                                                

 % curl 'localhost:8000/?item=1&event='
{"event":"","item":1}%                                                

 % curl 'localhost:8000/?item=1&event=1'
{"event":1,"item":1}%   
```

**Comment by jlmllorente** [comment]
> If the query parameter has no value, it can be cast as an empty string `""`. See the following example:
> 
> ```python
> from fastapi import FastAPI, Query, Request
> 
> app = FastAPI()
> 
> 
> @app.get("/")
> async def root(r: Request, event: int | str | None = None, item: int | None = None):
>     print(f"Event:{event}.")
>     print(f"Item:{item}.")
>     return {"event": event, "item": item}
> 
> 
> if __name__ == "__main__":
>     import uvicorn
> 
>     uvicorn.run(app, host="0.0.0.0", port=8000)
> ```
> 
> Following requests with responses (note the null value!):
> 
> ```shell
>  % curl 'localhost:8000/?item=1'
> {"event":null,"item":1}%                          
>                     
>  % curl 'localhost:8000/?item=1&event'
> {"event":"","item":1}%                                                
> 
>  % curl 'localhost:8000/?item=1&event='
> {"event":"","item":1}%                                                
> 
>  % curl 'localhost:8000/?item=1&event=1'
> {"event":1,"item":1}%   
> ```

This example doesn't make use of Request?
I'm also not trying to return the dictionary.
Apologies if I can't post the exact code I'm working but the use-cases I'm trying to work on is:
1. Return all events if ( no event is passed):
<img width="793" alt="image" src="https://user-images.githubusercontent.com/104721284/192057534-dfd9991e-4e94-4f5b-9fe0-5cf95cebc98e.png">

2. Raise an error message when message when event is passed but no value: 
![image](https://user-images.githubusercontent.com/104721284/192057697-72506d05-1a53-4612-9dcd-ef2bbb8ad9b2.png)

For some reason, when I do `params = request.query_params` the keys with blank values disappears in the request.

**Comment by JarroVGIT** [comment]
> This example doesn't make use of Request?

Correct, you don't need `Request`, you just need to define the parameter in your path operation function (like I did with `event: int | str | None = None`. Then in your function, your can check if `event==None` (then it didn't come as a query param), or `event==''` (then event came as a query param but without value) and else event has a value. 

I don't know why Postman is not doing it, maybe it is omitting empty query params in the actual call to the endpoint? You can check in the console of Postman. (I don't expect that to be the case though, but it's the only thing I can think of).

If I misunderstood your question, please provide a reproducible example that illustrates your issue in full.

**Comment by tiangolo** [comment]
Thanks for the help here @JarroVGIT ! 👏  🙇 

If that solves the original problem, then you can close this issue @jlmllorente  ✔️
